packages/runtime-core/src/componentRenderUtils.ts

```diff
@@ -248,13 +248,14 @@ export function renderComponentRoot(
   }
   // inherit transition data
   if (vnode.transition) {
-    if (__DEV__ && !isElementRoot(root)) {
+    const child = getInnerChild(root) ?? root
```

Comment:
```suggestion
    const child = getInnerChild(root) || root
```
see [generated code](https://www.babeljs.cn/repl#?browsers=safari%20%3E%3D%209%2Cchrome%3E%3D23&build=&builtIns=usage&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBAhjAvDAjAKFJWAjJ8YD8BqG40MweCAPtakA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Creact&prettier=false&targets=&version=7.25.6&externalPlugins=&assumptions=%7B%7D)